packages/Utils/src/DTK_Core.cpp

```diff
@@ -52,6 +49,35 @@ void initKokkos( int *argc, char ***argv )
                                         " is not initialized. Please report"
                                         " this bug to the DTK developers." );
 }
+
+// Initialize Kokkos, if it needs initialization.
+void initKokkos( const Kokkos::InitArguments &args )
+{
+    if ( !dtkInitializedKokkos )
+    {
+        // Kokkos doesn't have a global is_initialized().  However,
+        // Kokkos::initialize() always initializes the default execution
+        // space, so it suffices to check whether that was initialized.
+        const bool kokkosIsInitialized =
+            Kokkos::DefaultExecutionSpace::is_initialized();
+
+        if ( !kokkosIsInitialized )
+        {
+            // Unlike MPI_Init, Kokkos promises not to modify argc and argv.
+            Kokkos::initialize( args );
+            dtkInitializedKokkos = true;
+        }
+    }
+
+    const bool kokkosIsInitialized =
+        Kokkos::DefaultExecutionSpace::is_initialized();
+
+    if ( !kokkosIsInitialized )
+        throw DataTransferKitException( "At the end of initKokkos, Kokkos"
+                                        " is not initialized. Please report"
+                                        " this bug to the DTK developers." );
+}
```

Comment:
This is almost copy/paste from `initKokkos( int *argc, char ***argv ` isn't it?
Could you not use a variadic template and forward the arguments to `Kokkos::initialize( ... )`?